### DISCUSSION TITLE
Printing cost warning on `azd up` for apps initialized from our sample templates

### DISCUSSION THREAD

**Discussion by anfibiacreativa** [Azure/azure-dev#4224] [Ideas]
This is a feature request to improve the DX when initializing templates from our collections.

A flag could be added at config level, eg: `AZURE_FROM_COLLECTION=true` retrieving services from deployment plan to print a warning on costs and a link back to the calculator or a page with costs information, when running `azd up`.

`You will be deploying 
Service A,
Service B
in {standard/premium} tier, which will generate costs. For more information, [link to calculator](#)`

**Comment by kristenwomack** [comment]
I suggest we start small. In the terminal output where we link to the portal deployment, we could add a link to the cost estimator for the app in the portal. This creates a pointer to a solution that already exists without adding complexity. 

We _could_ output the services that are configured for the template, but adding the service and the tier, is still asking folks to calculate, but I question the value of putting it in two places and the complexity/cogitative cost, when we could point to the solution in the portal that exists.

**Reply by pamelafox** [reply]
Would developers provide the link to the cost estimator, or would you somehow generate a link to it?=

**Reply by anfibiacreativa** [reply]
> I question the value of putting it in two places and the complexity/cogitative cost, when we could point to the solution in the portal that exists.

I believe the value is 
* staying in context
* concrete information for the specific template, even if the readme fails to list a service

**Reply by kristenwomack** [reply]
@pamelafox - for projected costs after deployment, generating the link to the portal where the cost projection works. for a cost estimator before deployment, it depends on the input capabilities of the calculator.

@anfibiacreativa - I agree staying in context is valuable and ideal, and it becomes less valuable if we can't maintain giving incorrect cost is wrong. As switching out services and models, putting the services in the README can become brittle and incorrect. For this reason, I'd push more for linking to source of truth (the portal or the estimator).
